hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetUtils.java

```diff
@@ -32,6 +32,8 @@
  */
 public final class NetUtils {
   public static final Logger LOG = LoggerFactory.getLogger(NetUtils.class);
+  private static final Object EXCLUDE_NODES_LOCK = new Object();
```

Comment:
I think moving `removeOutOfScope` to `NetworkTopologyImpl` and making it non-static would be more future-proof.  Syncing on a static object seems risky.

Reply:
I agree that we could move this out, since only 1 caller. Why sync on static final risky?

Reply:
> Why sync on static final risky?

I think currently there is a single instance of `NetworkTopologyImpl` in SCM.  But you never know when another instance is needed, and then the lock will be a source of unnecessary contention.  I think it's safer to keep the lock instance-based.

Reply:
Hi, @adoroszlai actually the whole locking thing is unnecessary, the Collection is created method local and modified in place. The caller also documents this fact. Making a change to remove this as a utility method.

Reply:
Nice.